### ISSUE TITLE
lazyload youtube videos and investigate youtube embeds optimizations

### ISSUE THREAD

**Issue by selul** [Codeinwp/optimole-wp#253]
**Is your feature request related to a problem? Please describe.**
We can investigate a way of lazy loading the youtube videos embedded in a page.

This can be a toggle setting inside the lazyload tab.

**Comment by ineagu** [comment]
@selul we can run those benchmarks with wp-rocket as well or other plugin : https://sabrinazeidan.com/embed-youtube-video-wordpress-without-slowing/

**Comment by selul** [comment]
@GrigoreMihai we need also unit tests for this + a benchmark report based on this https://sabrinazeidan.com/embed-youtube-video-wordpress-without-slowing/

**Comment by GrigoreMihai** [comment]
I've added benchmark results here, benchmark was run on a page with 1 embedded iframe web page and 5 embedded videos, with the iframe lazy option off and on

https://photos.app.goo.gl/dKiphU27nZnHYste9

**Comment by selul** [comment]
@GrigoreMihai can we replicate the same test examples ( all 3 of them ) and compare with the same scenarios ( Optimole/wprocket/standalone as well as other youtube lazyload plugins )

**Comment by GrigoreMihai** [comment]
Yes, I have added the results here https://docs.google.com/document/d/1jwPPUiJo9z8qOcwkFD7_A0EDW0YG4E1l_oypTDrrtSU/edit?usp=sharing , the overall conclusion is that we have better speed than all tested plugins
